stdlib/src/utils/static_tuple.mojo

```diff
@@ -264,43 +264,74 @@ struct InlineArray[
     # Life cycle methods
     # ===------------------------------------------------------------------===#
 
-    @always_inline
-    fn __init__(inout self):
-        """This constructor will always cause a compile time error if used.
-        It is used to steer users away from uninitialized memory.
+    fn __init__(
+        inout self,
+        *,
+        owned unsafe_assume_initialized: InlineArray[
+            UnsafeMaybeUninitialized[Self.ElementType], Self.size
+        ],
+    ):
+        """Constructs an `InlineArray` from an `InlineArray` of `UnsafeMaybeUninitialized`.
+
+        Calling this function assumes that all elements in the input array are initialized.
+
+        If the elements of the input array are not initialized, the behavior is undefined,
+        even  if `ElementType` is valid *for every possible bit pattern* (e.g. `Int` or `Float`).
+
+        Args:
+            unsafe_assume_initialized: The array of `UnsafeMaybeUninitialized` elements.
         """
-        constrained[
-            False,
-            (
-                "Initialize with either a variadic list of arguments, a default"
-                " fill element or pass the keyword argument"
-                " 'unsafe_uninitialized'."
-            ),
-        ]()
+
         self._array = __mlir_op.`kgen.undef`[_type = Self.type]()
```

Comment:
Is this a temporary way to implement this? This function should just be a transmute (unsafe type cast).

Reply:
I think in the future we may be able to avoid the copy, but at the moment, it's the only way possible as far as I know.